### PR TITLE
Fix receiver name in client and comments

### PR DESCRIPTION
**Pull request by crashiura**
This was a trivial fix so there is no related Issue
I added comments and fixed receiver name in client.

### CODE REVIEW

File: govcd/admincatalog.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func (adminCatalog *AdminCatalog) Update() error {
 	return err
 }
 
-// Uploads an ova file to a catalog. This method only uploads bits to vCD spool area.
+// UploadOvf Uploads an ova file to a catalog. This method only uploads bits to vCD spool area.
```
Comment (Didainius) [inline govcd/admincatalog.go:62]:
```suggestion
// UploadOvf uploads an ova file to a catalog. This method only uploads bits to vCD spool area.
```

File: govcd/api.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func debugShowResponse(resp *http.Response, body []byte) {
 	}
 }
 
-// Convenience function, similar to os.IsNotExist that checks whether a given error
+// IsNotFound Convenience function, similar to os.IsNotExist that checks whether a given error
```
Comment (Didainius) [inline govcd/api.go:139]:
```suggestion
// IsNotFound is a convenience function, similar to os.IsNotExist that checks whether a given error
```

File: govcd/api.go [hunk 3]
```diff
@@ -379,7 +379,7 @@ func checkRespWithErrType(bodyType types.BodyType, resp *http.Response, err, err
 	}
 }
 
-// Helper function creates request, runs it, checks response and parses task from response.
+// ExecuteTaskRequest Helper function creates request, runs it, checks response and parses task from response.
```
Comment (Didainius) [inline govcd/api.go:382]:
```suggestion
// ExecuteTaskRequest helper function creates request, runs it, checks response and parses task from response.
```

File: govcd/api.go [hunk 4]
```diff
@@ -390,7 +390,7 @@ func (client *Client) ExecuteTaskRequest(pathURL, requestType, contentType, erro
 	return client.executeTaskRequest(pathURL, requestType, contentType, errorMessage, payload, client.APIVersion)
 }
 
-// Helper function creates request, runs it, checks response and parses task from response.
+// ExecuteTaskRequestWithApiVersion Helper function creates request, runs it, checks response and parses task from response.
```
Comment (Didainius) [inline govcd/api.go:393]:
```suggestion
// ExecuteTaskRequestWithApiVersion helper function creates request, runs it, checks response and parses task from response.
```

File: govcd/api.go [hunk 5]
```diff
@@ -436,7 +436,7 @@ func (client *Client) executeTaskRequest(pathURL, requestType, contentType, erro
 	return *task, nil
 }
 
-// Helper function creates request, runs it, checks response and do not expect any values from it.
+// ExecuteRequestWithoutResponse Helper function creates request, runs it, checks response and do not expect any values from it.
```
Comment (Didainius) [inline govcd/api.go:439]:
```suggestion
// ExecuteRequestWithoutResponse helper function creates request, runs it, checks response and do not expect any values from it.
```

File: govcd/api.go [hunk 6]
```diff
@@ -447,7 +447,7 @@ func (client *Client) ExecuteRequestWithoutResponse(pathURL, requestType, conten
 	return client.executeRequestWithoutResponse(pathURL, requestType, contentType, errorMessage, payload, client.APIVersion)
 }
 
-// Helper function creates request, runs it, checks response and do not expect any values from it.
+// ExecuteRequestWithoutResponseWithApiVersion Helper function creates request, runs it, checks response and do not expect any values from it.
```
Comment (Didainius) [inline govcd/api.go:450]:
```suggestion
// ExecuteRequestWithoutResponseWithApiVersion helper function creates request, runs it, checks response and do not expect any values from it.
```

File: govcd/api.go [hunk 7]
```diff
@@ -491,7 +491,7 @@ func (client *Client) executeRequestWithoutResponse(pathURL, requestType, conten
 	return nil
 }
 
-// Helper function creates request, runs it, check responses and parses out interface from response.
+// ExecuteRequest Helper function creates request, runs it, check responses and parses out interface from response.
```
Comment (Didainius) [inline govcd/api.go:494]:
```suggestion
// ExecuteRequest helper function creates request, runs it, check responses and parses out interface from response.
```

File: govcd/api.go [hunk 8]
```diff
@@ -504,7 +504,7 @@ func (client *Client) ExecuteRequest(pathURL, requestType, contentType, errorMes
 	return client.executeRequest(pathURL, requestType, contentType, errorMessage, payload, out, client.APIVersion)
 }
 
-// Helper function creates request, runs it, check responses and parses out interface from response.
+// ExecuteRequestWithApiVersion Helper function creates request, runs it, check responses and parses out interface from response.
```
Comment (Didainius) [inline govcd/api.go:507]:
```suggestion
// ExecuteRequestWithApiVersion helper function creates request, runs it, check responses and parses out interface from response.
```

File: govcd/api.go [hunk 9]
```diff
@@ -598,7 +598,10 @@ func executeRequestWithApiVersion(pathURL, requestType, contentType string, payl
 
 // executeRequestCustomErr performs request and unmarshals API error to errType if not 2xx status was returned
 func executeRequestCustomErr(pathURL string, params map[string]string, requestType, contentType string, payload interface{}, client *Client, errType error, apiVersion string) (*http.Response, error) {
-	url, _ := url.ParseRequestURI(pathURL)
+	requestURI, err := url.ParseRequestURI(pathURL)
+	if err != nil {
+		return nil, fmt.Errorf("could parse path requestURI %s", err)
```
Comment (Didainius) [inline govcd/api.go:603]:
```suggestion
		return nil, fmt.Errorf("could parse path request URI '%s': %s", pathURL, err)
```

File: govcd/vm.go [hunk 10]
```diff
@@ -34,7 +34,7 @@ func NewVM(cli *Client) *VM {
 	}
 }
 
-// create instance with reference to types.QueryResultVMRecordType
+// NewVMRecord create instance with reference to types.QueryResultVMRecordType
```
Comment (Didainius) [inline govcd/vm.go:37]:
```suggestion
// NewVMRecord creates an instance with reference to types.QueryResultVMRecordType
```

File: govcd/vm.go [hunk 11]
```diff
@@ -213,15 +213,15 @@ func (vm *VM) PowerOff() (Task, error) {
 		"", "error powering off VM: %s", nil)
 }
 
-// Sets number of available virtual logical processors
+// ChangeCPUCount Sets number of available virtual logical processors
```
Comment (Didainius) [inline govcd/vm.go:216]:
```suggestion
// ChangeCPUCount sets number of available virtual logical processors
```

File: govcd/vm.go [hunk 12]
```diff
@@ -213,15 +213,15 @@ func (vm *VM) PowerOff() (Task, error) {
...
 // (i.e. CPUs x cores per socket)
 // Cpu cores count is inherited from template.
 // https://communities.vmware.com/thread/576209
 func (vm *VM) ChangeCPUCount(virtualCpuCount int) (Task, error) {
 	return vm.ChangeCPUCountWithCore(virtualCpuCount, nil)
 }
 
-// Sets number of available virtual logical processors
+// ChangeCPUCountWithCore Sets number of available virtual logical processors
```
Comment (Didainius) [inline govcd/vm.go:224]:
```suggestion
// ChangeCPUCountWithCore sets number of available virtual logical processors
```

File: govcd/vm.go [hunk 13]
```diff
@@ -522,7 +522,7 @@ func (vm *VM) attachOrDetachDisk(diskParams *types.DiskAttachOrDetachParams, rel
 		attachOrDetachDiskLink.Type, "error attach or detach disk: %s", diskParams)
 }
 
-// Attach an independent disk
+// AttachDisk Attach an independent disk
```
Comment (Didainius) [inline govcd/vm.go:525]:
```suggestion
// AttachDisk attaches an independent disk
```

File: govcd/vm.go [hunk 14]
```diff
@@ -539,7 +539,7 @@ func (vm *VM) AttachDisk(diskParams *types.DiskAttachOrDetachParams) (Task, erro
 	return vm.attachOrDetachDisk(diskParams, types.RelDiskAttach)
 }
 
-// Detach an independent disk
+// DetachDisk Detach an independent disk
```
Comment (Didainius) [inline govcd/vm.go:542]:
```suggestion
// DetachDisk detaches an independent disk
```

File: govcd/vm.go [hunk 15]
```diff
@@ -556,7 +556,7 @@ func (vm *VM) DetachDisk(diskParams *types.DiskAttachOrDetachParams) (Task, erro
 	return vm.attachOrDetachDisk(diskParams, types.RelDiskDetach)
 }
 
-// Helper function which finds media and calls InsertMedia
+// HandleInsertMedia Helper function which finds media and calls InsertMedia
```
Comment (Didainius) [inline govcd/vm.go:559]:
```suggestion
// HandleInsertMedia helper function finds media and calls InsertMedia
```

File: govcd/vm.go [hunk 16]
```diff
@@ -617,7 +617,7 @@ func isMediaInjected(items []*types.VirtualHardwareItem) bool {
 	return false
 }
 
-// Helper function which finds media and calls EjectMedia
+// HandleEjectMedia Helper function which finds media and calls EjectMedia
```
Comment (Didainius) [inline govcd/vm.go:620]:
```suggestion
// HandleEjectMedia is a helper function which finds media and calls EjectMedia
```

File: govcd/vm.go [hunk 17]
```diff
@@ -638,7 +638,7 @@ func (vm *VM) HandleEjectMedia(org *Org, catalogName, mediaName string) (EjectTa
 	return task, err
 }
 
-// Insert media for VM
+// InsertMedia Insert media for VM
```
Comment (Didainius) [inline govcd/vm.go:641]:
```suggestion
// InsertMedia inserts media for a VM
```

File: govcd/vm.go [hunk 18]
```diff
@@ -651,7 +651,7 @@ func (vm *VM) InsertMedia(mediaParams *types.MediaInsertOrEjectParams) (Task, er
 	return vm.insertOrEjectMedia(mediaParams, types.RelMediaInsertMedia)
 }
 
-// Eject media from VM
+// EjectMedia Eject media from VM
```
Comment (Didainius) [inline govcd/vm.go:654]:
```suggestion
// EjectMedia ejects media from VM
```

File: govcd/vm.go [hunk 19]
```diff
@@ -746,7 +745,7 @@ func (vm *VM) GetQuestion() (types.VmPendingQuestion, error) {
 
 }
 
-// Use the provide answer to existing VM question for operation which need additional response
+// AnswerQuestion Use the provide answer to existing VM question for operation which need additional response
```
Comment (Didainius) [inline govcd/vm.go:748]:
```suggestion
// AnswerQuestion uses the provided answer to existing VM question for operation which need additional response
```

File: govcd/vm.go [hunk 20]
```diff
@@ -709,7 +709,7 @@ func (vm *VM) insertOrEjectMedia(mediaParams *types.MediaInsertOrEjectParams, li
 		insertOrEjectMediaLink.Type, "error insert or eject media: %s", mediaParams)
 }
 
-// Use the get existing VM question for operation which need additional response
+// GetQuestion Use the get existing VM question for operation which need additional response
```
Comment (Didainius) [inline govcd/vm.go:712]:
```suggestion
// GetQuestion uses the get existing VM question for operation which need additional response
```

**Review by Didainius — COMMENTED** [review]
Thanks for this PR.

Just a question - did you run  any test on it. I saw there were a few shuffles on error handling and wondering if it has any consequences.

**Comment by crashiura** [comment]
Thank you for comments, I ran unit tests

**Review by Didainius — APPROVED** [review]
Looks like it does not impacts tests after run.
Thanks.

Also - please do add a comment to `CHANGELOG.md` before other team members review it.

**Comment by crashiura** [comment]
I added comment in changelog.

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,5 +1,7 @@
 ## 2.12.0 (unreleased)
 
+* Improved error handling and fix receiver name in client
```
Comment (lvirbalas) [inline CHANGELOG.md:3]:
```suggestion
* Improved error handling and function receiver name in client
```

**Review by lvirbalas — APPROVED** [review]
LGTM, thanks for contributing.

**Comment by crashiura** [comment]
@dataclouder check pls

File: govcd/api.go [hunk 22]
```diff
@@ -598,7 +598,10 @@ func executeRequestWithApiVersion(pathURL, requestType, contentType string, payl
 
 // executeRequestCustomErr performs request and unmarshals API error to errType if not 2xx status was returned
 func executeRequestCustomErr(pathURL string, params map[string]string, requestType, contentType string, payload interface{}, client *Client, errType error, apiVersion string) (*http.Response, error) {
-	url, _ := url.ParseRequestURI(pathURL)
+	requestURI, err := url.ParseRequestURI(pathURL)
+	if err != nil {
+		return nil, fmt.Errorf("could parse path request URI '%s': %s", pathURL, err)
```
Comment (dataclouder) [inline govcd/api.go:603]:
```suggestion
		return nil, fmt.Errorf("couldn't parse path request URI '%s': %s", pathURL, err)
```

File: govcd/vm.go [hunk 23]
```diff
@@ -720,17 +720,16 @@ func (vm *VM) GetQuestion() (types.VmPendingQuestion, error) {
 	req := vm.client.NewRequest(map[string]string{}, http.MethodGet, *apiEndpoint, nil)
 
 	resp, err := vm.client.Http.Do(req)
+	if err != nil {
+		return types.VmPendingQuestion{}, fmt.Errorf("error getting question: %s", err)
```
Comment (dataclouder) [inline govcd/vm.go:724]:
```suggestion
		return types.VmPendingQuestion{}, fmt.Errorf("error getting VM question: %s", err)
```

**Review by dataclouder — CHANGES_REQUESTED** [review]

**Comment by crashiura** [comment]
@dataclouder done

**Comment by crashiura** [comment]
@dataclouder Hi, check pls

**Review by dataclouder — APPROVED** [review]

**Pull request merged**